packages/backend/src/workers/action.ts

```diff
@@ -143,6 +143,17 @@ worker.on('failed', async (job, err) => {
   }
 })
 
+worker.on('error', (err) => {
+  if (!err) {
+    logger.error('Worker undefined error')
+    return
+  }
+  // catch-all just in case any errors bubble up and potentially crash the worker task
+  logger.error(`Worker errored with ${err.message}`, {
+    err: err.stack,
+  })
+})
```

Comment:
paranoia: want to try catch this too so that worker don't die if logger error, since we already doing it for `uncaughtException`

```
```suggestion
try {
  if (!err) {
    logger.error('Worker undefined error')
    return
  }
  // catch-all just in case any errors bubble up and potentially crash the worker task
  logger.error(`Worker errored with ${err.message}`, {
    err: err.stack,
  })
})
}  catch (e) {
    // eslint-disable-next-line no-console
    console.log('Error happened to logger in worker')
    console.error(e)
  }
```
```

Reply:
can differentiate the workers also?
`console.log('Error happened to logger in action worker')`